website/docs/experimental/callbacks.md

```diff
@@ -0,0 +1,185 @@
+---
+id: callbacks
+title: Callbacks
+sidebar_label: Callbacks
+---
+
+import GithubLink from "@site/src/components/GithubLink"
+
+
+The <GithubLink to="hydra/experimental/callback.py">Callback interface</GithubLink> enables custom
+code to be triggered by various Hydra events.
+
+To use the callback API, one should import Hydra's `Callback` class:
+```python
+from hydra.experimental.callback import Callback
+```
+Users can then create subclasses of this `Callback` class, overriding one or more of
+the methods defined by `Callback`. For the methods of a subclass to be called at the
+appropriate time, the subclass must be registered with Hydra in the `hydra.callbacks` config
+ (see examples below).
+
+The full API exposed by the `hydra.experimental.callback.Callback` class is listed below:
+
+<details><summary>Events supported (Click to expand)</summary>
+
+```python
+class Callback:
+    def on_run_start(self, config: DictConfig, **kwargs: Any) -> None:
+        """
+        Called in RUN mode before job starts.
+        """
+        ...
+
+    def on_run_end(self, config: DictConfig, **kwargs: Any) -> None:
+        """
+        Called in RUN mode after job ends.
+        """
+        ...
+
+    def on_multirun_start(self, config: DictConfig, **kwargs: Any) -> None:
+        """
+        Called in MULTIRUN mode before any job starts.
+        """
+        ...
+
+    def on_multirun_end(self, config: DictConfig, **kwargs: Any) -> None:
+        """
+        Called in MULTIRUN mode after all jobs end.
+        """
+        ...
+
+    def on_job_start(self, config: DictConfig, **kwargs: Any) -> None:
+        """
+        Called in both RUN and MULTIRUN modes, once for each Hydra job (before running
+        application code).
+        """
+        ...
+
+    def on_job_end(
+        self, config: DictConfig, job_return: JobReturn, **kwargs: Any
+    ) -> None:
+        """
+        Called in both RUN and MULTIRUN modes, once for each Hydra job (after running
+        application code).
+        """
+        ...
+
+```
+</details>
+
+### Configure Callback
+
+Say we have `MyCallback` so after every job ends we can upload a certain file to a S3 bucket.
+For simplicity we include this Callback class within the application, in real life you should have the
+Callback in a separate file.
+Running the application, we can see our custom method `on_job_end` was called.
```

Comment:
This sentence does not read right to me.
Something like this would work better in my opinion:

In this example, we will use a callback to upload a certain file to an S3 bucket at the end of the job.
For simplicity we include this Callback class within the application, real applications are likely to place it in a different Python file.
Running the application, we can see our custom method `on_job_end` was called.